### DISCUSSION TITLE
An error was reported：x must be strictly increasing if s = 0

### DISCUSSION THREAD

**Discussion by 0batsy0** [phoebe-project/phoebe2#772] [Questions]
When I set all parameter and want to run mcmc,I got this.I don't understand what caused this error.
![捕获](https://github.com/phoebe-project/phoebe2/assets/130527283/6c448e47-78d7-4027-9806-65414f5ddabf)

**Comment by kecnry** [comment]
Do your initializing distributions successfully run a forward model on their own and are the resulting log probabilities finite?

You can check this with something like (sampling over the input distributions N times, running the forward model, and printing both the lnlikelihood and lnpriors (if you've set priors):

```
for i in range(50):
    tes.sample_distribution_collection('init_from@emcee_solver', set_value=True)
    tes.run_compute(model='tmp_model')
    lnlikelihood = tes.calculate_lnlikelihood(model='tmp_model')
    lnpriors = tes.calculate_lnp('priors@emcee_solver')
    print(f"{i}, lnlikelihood={lnlikelihood}, lnpriors={lnpriors}")
```

**Reply by 0batsy0** [reply]
After running for a period of time, I received this error message：
![捕获](https://github.com/phoebe-project/phoebe2/assets/130527283/351b8ac0-c4ca-4041-ba25-a38bfa7dcf8e)
I don't quite understand how this error occurred and how I should modify my code to make it run properly

**Reply by kecnry** [reply]
Some combination of your input parameters were drawn outside the bounds of the atmosphere table.  For mcmc this isn't a huge deal if it happens occasionally, as these will just be rejected iterations (as long as it can find enough successful starting positions for the walkers), but you can also avoid it by tightening your initializing distributions and/or adding priors.

**Comment by 0batsy0** [comment]
Thank you for your reply.I‘m not sure my code is right.But for now,whether I use phoebe.gaussian or phoebe.uniform,my initializing distributions successfully run a forward model on their own.
And when I use phoebe.gaussian, Its output value will change and it can operate MCMC normally.
![2](https://github.com/phoebe-project/phoebe2/assets/130527283/65d10406-204d-4dc1-aba3-0d85a05322bf)
But when I use phoebe.uniform,the lnpriors value it outputs does not change, and there may be occasional inf values.At the same time, if you run mcmc, you will receive an error message:x must be strictly increasing if s = 0.
![1](https://github.com/phoebe-project/phoebe2/assets/130527283/7bad9dc8-a0cd-4964-9817-49b3719f7ebf)
I don't quite understand what caused this.

**Reply by kecnry** [reply]
I would not suggest using the same distribution as your sampling/initializing distribution and your priors.  In general, your initializing distributions should be pretty constrained around what you think is the correct solution, and your priors should only be information that you _know_ from other sources or perhaps to limit the model from wandering into areas of parameter space that you know should be excluded or are unphysical.  I'm not sure if this is the cause of the problem or not, but its possible that all walkers were getting -infs and didn't know how to recover.
